core/sql/optimizer/GroupAttr.cpp

```diff
@@ -327,6 +327,38 @@ void GroupAttributes::addConstraint(ItemExpr *c)
       }
       break;
 
+    case ITM_CHECK_OPT_CONSTRAINT:
+      {
+        CheckOptConstraint *cc = (CheckOptConstraint *) c;
+
+        // check existing uniqueness constraints whether they are similar
+        // and combine uniqueness constraints if possible
+        for (ValueId occ = constraints_.init();
+             constraints_.next(occ);
+             constraints_.advance(occ))
+          {
+            if (occ.getItemExpr()->getOperatorType() ==
+		ITM_CHECK_OPT_CONSTRAINT)
+              {
+                const ValueIdSet &occPreds =
+                  ((CheckOptConstraint *) occ.getItemExpr())->getCheckPreds();
+
+                if (occPreds.contains(cc->getCheckPreds()))
```

Comment:
Sorry, just realize that below, the code may add the same check constraint multiple times, so we do need the simple guard against that. Therefore the check is not nearly useless, it's just not as smart as it could be.